File: atlan-performance-mobile/shared/src/commonMain/kotlin/com/atlan/performance/shared/design/AtlanColors.kt
package com.atlan.performance.shared.design

/**
 * Shared color tokens as 0xAARRGGBB longs. Platform layers map these to native color types.
 * Source of truth: DESIGN_TOKENS.md.
 *
 * Discipline: Coral is rare (productive action / completion / high-signal accents only). Tide marks
 * science, calm progress, selected state, and Why affordances. Never use Coral or red for
 * errors, guilt, or missed-session states.
 */
object AtlanColors {
    const val Abyss: Long = 0xFF0B2A3C
    const val AbyssDeep: Long = 0xFF061A26
    const val Tide: Long = 0xFF0E8A9A
    const val TideDeep: Long = 0xFF0A6F7D
    const val TideSoft: Long = 0xFFBFE0E5
    const val TidePale: Long = 0xFFDDEEF1
    const val Coral: Long = 0xFFFF6A3D
    const val CoralBright: Long = 0xFFFF7E50
    const val CoralDeep: Long = 0xFFE55428
    const val Foam: Long = 0xFFECF7F8
    const val FoamWarm: Long = 0xFFF4FAFB
    const val Paper: Long = 0xFFFBFCFC
}

File: atlan-performance-mobile/iosApp/AtlanPerformance/Design/AtlanColors.swift
import SwiftUI

/// SwiftUI mirror of the shared color tokens (DESIGN_TOKENS.md).
/// Coral is rare (productive action / completion / high-signal accents). Tide marks science, calm
/// progress, selected state, and Why affordances. Never use Coral or red for disruption.
extension Color {
    init(hex: UInt32) {
        let r = Double((hex >> 16) & 0xFF) / 255.0
        let g = Double((hex >> 8) & 0xFF) / 255.0
        let b = Double(hex & 0xFF) / 255.0
        self.init(.sRGB, red: r, green: g, blue: b, opacity: 1.0)
    }
}

enum AtlanColors {
    static let abyss = Color(hex: 0x0B2A3C)
    static let abyssDeep = Color(hex: 0x061A26)
    static let tide = Color(hex: 0x0E8A9A)
    static let tideDeep = Color(hex: 0x0A6F7D)
    static let tideSoft = Color(hex: 0xBFE0E5)
    static let tidePale = Color(hex: 0xDDEEF1)
    static let coral = Color(hex: 0xFF6A3D)
    static let coralBright = Color(hex: 0xFF7E50)
    static let coralDeep = Color(hex: 0xE55428)
    static let foam = Color(hex: 0xECF7F8)
    static let foamWarm = Color(hex: 0xF4FAFB)
    static let paper = Color(hex: 0xFBFCFC)
}
